### ISSUE TITLE
Unable to start up MapGPT

### ISSUE THREAD

**Issue by cactusoftheday** [chen-judge/MapGPT#7]
I've been trying to start up MapGPT but haven't successfully done so. Here's what I've been doing:

1. Build Matterport3DSimulator
2. Enter the Matterport3D docker container
3. Download anaconda and create the environnment
4. Clone the repo
5. Download the datasets and unzip
6. Run scripts

How did you install numpy 1.20.3? It's not supported on python 3.10 and for some reason seems to give a very large error log when attempting an install. Fixed by installing numpy 1.22.x.

However, when I try to run `gpt4o.sh` I keep getting errors like `ModuleNotFoundError: No module named 'vln'`. Could you provide some clearer steps on how to reproduce your results? 

Fixed it by adding this to my PYTHONPATH environment variable: `export PYTHONPATH=$PYTHONPATH:~/mount/Matterport3DSimulator/MapGPT/vln:~/mount/Matterport3DSimulator/MapGPT/utils:~/mount/Matterport3DSimulator/MapGPT/GPT:~/mount/Matterport3DSimulator/MapGPT`

**Comment by chen-judge** [comment]
I'm very sorry that I confused two different environments that I have used. In the released Python 3.10 environment, I was actually using numpy 1.26.2. Thank you for helping me find this error. 

As for the second issue, it seems I haven't encountered it before. I didn't configure using Docker, but instead built directly, which may have caused some differences.

**Comment by wlll123456** [comment]
> I've been trying to start up MapGPT but haven't successfully done so. Here's what I've been doing:我一直在尝试启动 MapGPT，但没有成功。下面是我一直在做的事情：
> 
> 1. Build Matterport3DSimulator构建 Matterport3DS 模拟器
> 2. Enter the Matterport3D docker container输入 Matterport3D docker 容器
> 3. Download anaconda and create the environnment下载 anaconda 并创建环境
> 4. Clone the repo 克隆 repo
> 5. Download the datasets and unzip下载数据集并解压
> 6. Run scripts 运行脚本
> 
> How did you install numpy 1.20.3? It's not supported on python 3.10 and for some reason seems to give a very large error log when attempting an install. Fixed by installing numpy 1.22.x.你是如何安装 numpy 1.20.3 的？python 3.10 不支持该版本，而且由于某些原因，在尝试安装时似乎会出现大量错误日志。安装 numpy 1.22.x 后已修复。
> 
> However, when I try to run `gpt4o.sh` I keep getting errors like `ModuleNotFoundError: No module named 'vln'`. Could you provide some clearer steps on how to reproduce your results?但是，当我尝试运行 `gpt4o.sh` 时，却不断出现 `ModuleNotFoundError: No module named 'vln'` 这样的错误。您能否提供一些更清晰的步骤，说明如何重现您的结果？
> 
> Fixed it by adding this to my PYTHONPATH environment variable: `export PYTHONPATH=$PYTHONPATH:~/mount/Matterport3DSimulator/MapGPT/vln:~/mount/Matterport3DSimulator/MapGPT/utils:~/mount/Matterport3DSimulator/MapGPT/GPT:~/mount/Matterport3DSimulator/MapGPT`在我的 PYTHONPATH 环境变量中添加以下内容后，问题就解决了： `export PYTHONPATH=$PYTHONPATH:~/mount/Matterport3DSimulator/MapGPT/vln:~/mount/Matterport3DSimulator/MapGPT/utils:~/mount/Matterport3DSimulator/MapGPT/GPT:~/mount/Matterport3DSimulator/MapGPT`

Hi cactusoftheday, I’d also like to replicate MapGPT using Docker. Have you successfully managed to do so? If so, could you provide a detailed outline of your process? Also, may I add you as a contact for further collaboration?

**Comment by cactusoftheday** [comment]
I did manage to build it. I cloned the MapGPT and MatterSim repo repo on my machine first. It'll help if you clone the MapGPT repo into the mattersim repo. Then did all the mattersim setup too, that includes entering the docker container and building the simulator code (one of the steps). 

From there, I downloaded the RGB_Observations dataset and followed the MapGPT instructions pretty straight forwardly. When it came to running the shell script that's where I started to have problems.

1. To solve the numpy version problem, install numpy 1.22.x
2. Missing module h5py, just `pip install h5py`. The latest version currently works.
3. `ModuleNotFoundError: No module named 'vln'`: Run this command: `export PYTHONPATH=$PYTHONPATH:~/mount/Matterport3DSimulator/MapGPT/vln:~/mount/Matterport3DSimulator/MapGPT/utils:~/mount/Matterport3DSimulator/MapGPT/GPT:~/mount/Matterport3DSimulator/MapGPT`

If you've got any more questions, I can try to answer them. I'd love to collaborate with you in the future!
